### DISCUSSION TITLE
Tinkerboard S und S.USV tinker mobile

### DISCUSSION THREAD

**Discussion by akula3** [OpenCCU/OpenCCU#1122] [🇩🇪 Fragen&Antworten] [answered]
Hallo Jens und alle zusammen,

Ich versuche vergeblich die beiden zum Laufen aus dem Subject zu bringen:
```
root@Raspberrymatic:~# /opt/susvd/susv -status
Incompatible Hardware or Software!
root@Raspberrymatic:~# /etc/init.d/S51susvd restart
Stopping S.USV daemon: OK
Starting S.USV daemon: Kann Datenrichtung nicht setzen (write)!
: Invalid argument
WARNING: couldn't identify firmware version, OK
```
Habe auch versucht, die FW zu aktualisieren (einmal aus /opt/susvd/fw/ und einmal die von Olmatic direkt) - beides hat nichts gebracht:
```
root@Raspberrymatic:~# /opt/susvd/susv -flash /opt/susvd/fw/susv_fw_22_tinker_mobile.hex
device         : /dev/i2c-1       (address: 0x30)
version        : TWIBOOT m8v2.1 (sig: 0x1e 0x93 0x07 => AVR Mega 8)
flash size     : 0x1c00 /  7168   (0x40 bytes/page)
eeprom size    : 0x0200 /   512
writing flash  : [**************************************************] (6814)
```
Im susvd.log gib es folgendes:
```
Wed Feb 10 15:32:52 2021 - SUSVD: Set GPIO27 to input
Wed Feb 10 15:32:52 2021 - SUSVD: Entering Daemon SUSV.
Wed Feb 10 15:44:47 2021 - SUSVD:  Unable to change to output
Wed Feb 10 15:44:49 2021 - SUSVD:  Set GPIO27 to input
Wed Feb 10 15:44:49 2021 - SUSVD: Entering Daemon SUSV.
```
Im System Log:
```
Feb 10 15:32:52 Raspberrymatic user.debug kernel: [   66.089261] gpio gpiochip1: Persistence not supported for GPIO 3
Feb 10 15:32:52 Raspberrymatic user.err kernel: [   66.091161] rockchip-pinctrl pinctrl: pin 3 is unrouted
.
.
.
Feb 10 15:44:47 Raspberrymatic user.err kernel: [  781.238918] rockchip-pinctrl pinctrl: pin 3 is unrouted
Feb 10 15:44:49 Raspberrymatic user.debug kernel: [  782.775585] gpio gpiochip1: Persistence not supported for GPIO 3
Feb 10 15:44:49 Raspberrymatic user.err kernel: [  782.780621] rockchip-pinctrl pinctrl: pin 3 is unrouted
```
Im Homematic-forum gibt es zwar Leute mit gleichem Problem, aber keine Lösung für.
Könnt Ihr mir bitte helfen?

**Comment by akula3** [comment]
Ich habe mir Mal das Debian Packet für Tinker von Olmatic angeschaut.
Dort steht in der susv.cfg
gpio=166
anstatt
gpio=27
Habe dann versucht, den GPIO in 166 zu ändern - klappt aber nicht, weil "Read-only file system".

@jens-maus, wie kann ich die susv.cfg Datei bearbeiten?

Auch alle anderen Dateien in /opt/susvd und dadrunter weisen Unterschiede auf. Ist das schlimm?

**Reply by jens-maus** [reply]
> @jens-maus, wie kann ich die susv.cfg Datei bearbeiten?

Du musst vorher das rootfs eben read-write remounten. Und zwar via:

```
mount -o rw,remount /
```

Danach kannst du die Datei editieren und dann danach wieder als `ro` remounten um Änderungen wieder zu verhindern. Nach einem Firmware-Update werden diese Änderungen aber übrigens wieder verschwunden sein.

**Reply by akula3** [reply]
Danke für den Tipp, Jens!
Mit der Änderung des GPIO's läuft S.USV immernoch nicht, aber man hat keine Fehlermeldung "Unable to change to output" im susvd.log mehr.
Dann habe ich aus dem Debian Paket von Olmatic folgendes auf dem Tinkerboard ersetzt:
```
/opt/susvd/susv
/opt/susvd/susvd
/opt/susvd/tools/twiboot
```
Danach lief der Daemon nach dem Aufruf von `/opt/susvd/susvd -start`:
```
root@Raspberrymatic:/opt/susvd# /opt/susvd/susv -status
********************************
*                              *
* S.USV solutions              *
* www.s-usv.com                *
*                              *
* Model: Tinker mobile         *
* Firmware Version: 2.20       *
* Software Version: 2.40       *
* Hardware Version: 3.4        *
*                              *
* Mail notification: Disabled  *
*                              *
* Timed Boot: Disabled         *
* Boot time: 00:00:00          *
*                              *
* Timed Shutdown: Disabled     *
* Shutdown time: 00:00:00      *
*                              *
* Thu Feb 11 16:02:47 2021     *
*                              *
********************************
*                              *
* Powering Source: Primary     *
* Charging circuit: ONLINE     *
* Charging current: 300 mA     *
*                              *
* Voltage in: 5.12 V           *
* Battery capacity: 100.00%    *
* Battery voltage: 4.20V       *
* Power Battery: 000.00 mA     *
* Power Extern: 000.00 mA      *
*                              *
* Shutdown timer: -1           *
* Autostart:  disabled         *
* Sleep timer: 5               *
*                              *
********************************
```
Allerdings es läuft nicht mit `/etc/init.d/S51susvd restart`:
```
root@Raspberrymatic:/opt/susvd# /etc/init.d/S51susvd restart
Stopping S.USV daemon: OK
Starting S.USV daemon: ^C
```
Der Script bleibt hängen, und ich musste es mit STRG+C abbrechen.
Dann habe ich versucht, die FW 2.20 im S51susvd Script einzupflegen. Das hat nichts gebracht, keine Änderung.
Ich werde weiter versuchen, aber @jens-maus , hast Du vielleich eine Idee?

**Comment by akula3** [comment, marked as answer]
Es läuft soweit:
```
root@Raspberrymatic:/opt/susvd# /etc/init.d/S51susvd restart
Stopping S.USV daemon: OK
Starting S.USV daemon: 2.20, OK
```
```
/opt/susvd$ /opt/susvd/susv -status
********************************
*                              *
* S.USV solutions              *
* www.s-usv.com                *
*                              *
* Model: Tinker mobile         *
* Firmware Version: 2.20       *
* Software Version: 2.40       *
* Hardware Version: 3.4        *
*                              *
* Mail notification: Disabled  *
*                              *
* Timed Boot: Disabled         *
* Boot time: 00:00:00          *
*                              *
* Timed Shutdown: Disabled     *
* Shutdown time: 00:00:00      *
*                              *
* Fri Feb 12 14:31:21 2021     *
*                              *
********************************
*                              *
* Powering Source: Primary     *
* Charging circuit: ONLINE     *
* Charging current: 300 mA     *
*                              *
* Voltage in: 5.22 V           *
* Battery capacity: 94.65%     *
* Battery voltage: 4.14V       *
* Power Battery: 000.00 mA     *
* Power Extern: 427.00 mA      *
*                              *
* Shutdown timer: -1           *
* Autostart:  disabled         *
* Sleep timer: 1               *
*                              *
********************************
```

Wie habe ich es erreicht:

1. Mit `mount -o rw,remount /` Dateisystem für Änderungen freischalten.

2. Aus dem entsprechenden [Debian Paket für S.USV Client Software von Olmatic](https://s-usv.de/downloads/) folgendes auf dem Tinkerboard ersetzen:
```
/opt/susvd/susv
/opt/susvd/susvd
/opt/susvd/tools/twiboot
```

3. In der `/opt/susvd/susv.cfg` einstellen `gpio=166`

4. In der `/etc/init.d/S51susvd` unter `case` ein neuen Block einfügen:
```
            # v3.4x should be a S.USV tinker mobile with HW Version 3
            2.2*)
              DAEMON=/opt/susvd/susvd
            ;;
```
und beim `restart|force-reload)` den `sleep` auf 3 erhöhen, da ich den Eindruck habe, daß der S.USV etwas mehr an Zeit für die Antworten braucht.

5. Mit `mount -o ro,remount /` Dateisystem für Änderungen wieder sperren.



**Mit dieser Lösung werden höchstwahrscheinlich die HW Versionen 2 nicht laufen**, weil die Original Dateien `/opt/susvd/susv`, `/opt/susvd/susvd`, `/opt/susvd/tools/twiboot` von Olmatic sich von HW Version zu HW Version unterscheiden. Dann ist in dem Fall auch die `case` Auswahl in der `/etc/init.d/S51susvd` eigentlich überflüssig. Ich werde weiter für eine universelle Lösung, wie bei @jens-maus, probieren.

Vielen Dank nochmal an Jens für Raspberrymatic und seine Hilfe!
